spec/linked_list.js

```diff
@@ -0,0 +1,205 @@
+import chai, { expect } from 'chai'
+import chaiChange from 'chai-change'
+import LinkedList from '../src/linked_list'
+
+chai.use(chaiChange)
+
+describe('Linked List', () => {
+  'use strict'
+
+  it('exists', () => {
+    expect(LinkedList).to.be.a('function')
+  })
+
+  context('getHeadNode()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+    myLinkedList.insert('third')
+
+    it('Returns an object', () => {
+      expect(myLinkedList.getHeadNode())
+        .to.be.a.object
+    })
+    it('Returns the first node in the list', () => {
+      expect(myLinkedList.getHeadNode().data)
+      .to.eql('first')
+    })
+  })
+  context('getTailNode()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+    myLinkedList.insert('third')
+
+    it('Returns an object', () => {
+      expect(myLinkedList.getTailNode())
+        .to.be.a.object
+    })
+    it('Returns the first node in the list', () => {
+      expect(myLinkedList.getTailNode().data)
+      .to.eql('third')
+    })
+  })
+  context('contains()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+
+    it('returns true the set contains the element', () => {
+      expect(myLinkedList.contains('first'))
+      .to.be.true
+    })
+    it('returns true if the element is last in the set', () => {
+      expect(myLinkedList.contains('second'))
+      .to.be.true
+    })
+    it('returns false if the set does not contain the element', () => {
+      expect(myLinkedList.contains('third'))
+      .to.be.false
+    })
+  })
+  context('find()', () => {
+    const myLinkedList = new LinkedList()
+    myLinkedList.insert('first')
+    myLinkedList.insert('second')
+
+    it('Returns the first node containing the provided data', () => {
```

Comment:
this it statement provides a false sense of security. You should write your it statements in a way that is very very precise. for example, to test this it statement you would have to have TWO second nodes and make sure the it only returns to first (checking the .next property is a way to check where a node is in the list without using the data property.